src/extension/wps-download/src/main/java/org/geoserver/wps/gs/download/DownloadAnimationProcess.java

```diff
@@ -175,8 +193,19 @@ public DownloadAnimationProcess(
             AWTSequenceEncoder enc = new AWTSequenceEncoder(out, frameRate);
 
             DownloadServiceConfiguration configuration = confiGenerator.getConfiguration();
-            TimeParser timeParser = new TimeParser(configuration.getMaxAnimationFrames());
-            Collection parsedTimes = timeParser.parse(time);
+            int totalFrames;
+            Collection parsedTimes = null;
+            List<String> animateValuesList = new ArrayList<>();
+            if (!StringUtils.isEmpty(time)) {
+                TimeParser timeParser = new TimeParser(configuration.getMaxAnimationFrames());
+                parsedTimes = timeParser.parse(time);
+                totalFrames = parsedTimes.size();
+            } else if (!StringUtils.isEmpty(animateParam) && !StringUtils.isEmpty(animateValues)) {
+                animateValuesList = Arrays.asList(animateValues.split(","));
```

Comment:
Should give the possibilty to escape commas.
I _believe_ this code handled the escaping, but not 100% sure: https://github.com/geoserver/geoserver/blob/2.17.x/src/wms/src/main/java/org/geoserver/wms/animate/FrameCatalog.java#L51
If not that one, it may be in the vicinity (other classes of the same package, for example).